### PR TITLE
move the tool/bindings_generator dart+node script into the top-level bindings_generator/ directory

### PR DESCRIPTION
**Pull request by devoncarew**
- move the tool/bindings_generator dart+node script into the top-level `bindings_generator/` directory
- update the docs on regenerating the bindings

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by kevmoo** [comment]
...uh...I guess?

**Comment by devoncarew** [comment]
Well, this refactor isn't critical, but I think it does help to separate out the outer tool (`tool/update_bindings.dart`) - that you run from the cli - from the dart code that's compiled to JS and run as a node app (invoked from that initial tool). It helps clarify which dart files are compiled to JS and which aren't.

I have no issues closing the PR if you don't think its worth it; cc @kevmoo @srujzs.

File: tool/update_bindings.dart [hunk 1]
```diff
@@ -222,22 +222,19 @@ Future<void> _generateJsTypeSupertypes() async {
...
-  const Map<String, String?> jsTypeSupertypes = $jsTypeSupertypes;
-  ''';
+const Map<String, String?> jsTypeSupertypes = {
+${jsTypeSupertypes.entries.map((e) => "  ${e.key}: ${e.value},").join('\n')}
+};
+''';
   final jsTypeSupertypesPath =
       p.join(_bindingsGeneratorPath, 'js_type_supertypes.dart');
   await File(jsTypeSupertypesPath).writeAsString(jsTypeSupertypesScript);
-  await _runProc(
```
Comment (srujzs) [inline tool/update_bindings.dart:236]: Is the formatting done elsewhere?

**Comment by srujzs** [comment]
I don't mind, it's nice to only keep `update_bindings.dart` in `tools` so you know what file to run easily.

**Review by srujzs — APPROVED** [review]

File: tool/update_bindings.dart (hunk 1 above)
Comment (devoncarew) [reply tool/update_bindings.dart:236]: we're just emitting code now that's already formatted :)

**Pull request merged**